x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/ranges/range_editor.tsx

```diff
@@ -95,17 +111,36 @@ const BaseRangeEditor = ({
   onMaxBarsChange: (newMaxBars: number) => void;
   onChangeIncludeEmptyRows: (includeEmptyRows: boolean) => void;
   includeEmptyRows?: boolean;
+  customBounds?: ExtendedBounds;
+  onChangeCustomBounds: (newBounds?: Required<ExtendedBounds>) => void;
+  dataBounds: { min: number; max: number } | undefined;
 }) => {
-  const [maxBarsValue, setMaxBarsValue] = useState(String(maxBars));
-
-  useDebounceWithOptions(
-    () => {
+  const onChangeMaxBars = useCallback(
+    (maxBarsValue: string) => {
       onMaxBarsChange(Number(maxBarsValue));
     },
-    { skipFirstRender: true },
-    TYPING_DEBOUNCE_TIME,
-    [maxBarsValue]
+    [onMaxBarsChange]
+  );
+  const { inputValue: maxBarsValue, handleInputChange: setMaxBarsValue } =
```

Comment:
I've taken the opportunity here to refactor this with the `useDebounceValue` hook.